### ISSUE TITLE
Overlapping filter dropdown options

### ISSUE THREAD

**Issue by joverlee521** [nextstrain/auspice#1453]
The options within the Filter Data dropdown overlap when the strings are too long making it impossible to read the values. The current work-around would be to use the options listed in the collapsed filters at the bottom of the page instead of the dropdown menu. 

![Screen Shot 2022-02-09 at 5 27 05 PM](https://user-images.githubusercontent.com/40774762/153319432-ed6b20a7-08b4-401d-9024-57cd4bac8507.png)

**How to reproduce**  
First noticed when checking the [trial build](https://nextstrain.org/staging/ncov/gisaid/trial/add_authors/global) for ncov that added originating & submitting labs to the filter.

**Possible solution**  
We can add a custom styles to the dropdown menu to either
1. Make the option height fit the content and display the full string (makes the dropdown options less uniform)
2. Hide the overflowing text (maybe add hover to display the full string?)

**Comment by jameshadfield** [comment]
This has been bothering me for ages, it would be great to fix! I would go for (1), if it's easy? (as noted in your measurements panel, upgrading react-select would also be nice)

**Comment by corneliusroemer** [comment]
I've also noticed this and found it limiting. I didn't know or think about the workaround.

I bumped both priority and likelihood from 1 to 2 since everyone using filter will encounter this and it is quite limiting, not just messed up layout but hindering ability to use filters.

The presence of this issue also stops us from adding more useful color-bys (see https://github.com/nextstrain/ncov/pull/861#issuecomment-1034169442), because we don't want to exacerbate the bug.
